source/opt/mem_pass.cpp

```diff
@@ -755,7 +755,9 @@ void MemPass::RemovePhiOperands(
     i += 2;
   }
 
+  context()->ForgetUses(phi);
   phi->ReplaceOperands(keep_operands);
+  context()->AnalyzeUses(phi);
```

Comment:
Neat.  I think this is what's causing my segfaults in https://github.com/dnovillo/SPIRV-Tools/tree/instr-to-block